ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/runs/TimeFilter.tsx

```diff
@@ -0,0 +1,74 @@
+import RelativeTimeFilter from '@inngest/components/Filter/RelativeTimeFilter';
+import { type Option } from '@inngest/components/Select/Select';
+import { useQuery } from 'urql';
+
+import { graphql } from '@/gql';
+
+const GetBillingPlanDocument = graphql(`
+  query GetBillingPlan {
+    account {
+      plan {
+        id
+        name
+        features
+      }
+    }
+
+    plans {
+      name
+      features
+    }
+  }
+`);
+
+type RelativeTimeFilterProps = {
+  selectedDays: string;
+  onDaysChange: (value: string) => void;
+};
+
+const daysAgoArray = ['1', '3', '7', '14', '30'];
+
+export default function TimeFilter({ selectedDays, onDaysChange }: RelativeTimeFilterProps) {
+  const [{ data }] = useQuery({
+    query: GetBillingPlanDocument,
+  });
+
+  // Since "features" is a map, we can't be 100% sure that there's a log
+  // retention value. So default to 7 days.
+  let logRetention = 7;
+  if (typeof data?.account.plan?.features.log_retention === 'number') {
+    logRetention = data.account.plan.features.log_retention;
+  }
```

Comment:
Can we make a React context for this? There's a handful of places that need to know the plan's history

Reply:
Created a [Linear ticket ](https://linear.app/inngest/issue/INN-3005/add-plans-validation-as-context) for that one

Reply:
Yeah - an "Entitlements" context would be ideal!